### ISSUE TITLE
BigInt encoded in scientific notation

### ISSUE THREAD

**Issue by joroKr21** [circe/circe#1033]
https://scastie.scala-lang.org/CYDFcG1GSPSMxWqgsUi1Vw
```scala
import io.circe.syntax._

println(BigInt(1000).asJson)
// 1e3
```

Is this expected? Looks like a bug to me.

**Comment by plokhotnyuk** [comment]
@joroKr21 as W/A you can try the following custom encoder:
```
 implicit val bigIntEncoder: Encoder[BigInt] = Encoder.encodeJsonNumber
    .contramap(x => JsonNumber.fromDecimalStringUnsafe(new java.math.BigDecimal(x.bigInteger).toPlainString))
```
Also, please, see how it works in [a context of benchmarks](https://github.com/plokhotnyuk/jsoniter-scala/commit/f04d08293af696e27f93df9c9e095229ab122dfb) (results to be published in a week).

**Comment by joroKr21** [comment]
Thanks for the for the workaround. I didn't think it was possible to achieve!

**Comment by joroKr21** [comment]
@plokhotnyuk Btw why are you creating an intermidate `BigDecimal`? Why not simply:
```scala
implicit val bigIntEncoder: Encoder[BigInt] = Encoder.encodeJsonNumber
    .contramap(x => JsonNumber.fromDecimalStringUnsafe(x.toString))
```

**Comment by plokhotnyuk** [comment]
@joroKr21 both options suffer from O(n^2) efficiency of Java's serialization for big numbers, but 1st one is more efficient on small numbers. Below are result of the same benchmark where circe serializes `BigInt` with different numbers of decimal digits:
https://github.com/plokhotnyuk/jsoniter-scala/blob/master/jsoniter-scala-benchmark/src/main/scala/com/github/plokhotnyuk/jsoniter_scala/benchmark/BigIntBenchmark.scala#L58

For `x.toString` : 
```
[info] REMEMBER: The numbers below are just data. To gain reusable insights, you need to follow up on
[info] why the numbers are the way they are. Use profilers (see -prof, -lprof), design factorial
[info] experiments, perform baseline and negative tests that provide experimental control, make sure
[info] the benchmarking environment is safe on JVM/OS/HW level, ask for reviews from the domain experts.
[info] Do not assume the numbers tell you what you want them to tell.
[info] Benchmark                    (size)   Mode  Cnt         Score        Error  Units
[info] BigIntBenchmark.writeCirce        1  thrpt    5  13726558.385 ± 944768.965  ops/s
[info] BigIntBenchmark.writeCirce       10  thrpt    5  11267725.825 ± 203663.550  ops/s
[info] BigIntBenchmark.writeCirce      100  thrpt    5    591196.514 ±  35587.121  ops/s
[info] BigIntBenchmark.writeCirce     1000  thrpt    5     27958.893 ±   1037.729  ops/s
[info] BigIntBenchmark.writeCirce    10000  thrpt    5      1280.834 ±     40.318  ops/s
[info] BigIntBenchmark.writeCirce   100000  thrpt    5        50.818 ±      0.926  ops/s
[info] BigIntBenchmark.writeCirce  1000000  thrpt    5         1.800 ±      0.019  ops/s
```

For `new java.math.BigDecimal(x.bigInteger).toPlainString` : 
```
[info] REMEMBER: The numbers below are just data. To gain reusable insights, you need to follow up on
[info] why the numbers are the way they are. Use profilers (see -prof, -lprof), design factorial
[info] experiments, perform baseline and negative tests that provide experimental control, make sure
[info] the benchmarking environment is safe on JVM/OS/HW level, ask for reviews from the domain experts.
[info] Do not assume the numbers tell you what you want them to tell.
[info] Benchmark                    (size)   Mode  Cnt         Score         Error  Units
[info] BigIntBenchmark.writeCirce        1  thrpt    5  24967866.981 ± 1448762.454  ops/s
[info] BigIntBenchmark.writeCirce       10  thrpt    5  19213205.318 ±  492795.536  ops/s
[info] BigIntBenchmark.writeCirce      100  thrpt    5    738645.113 ±   38627.156  ops/s
[info] BigIntBenchmark.writeCirce     1000  thrpt    5     32598.228 ±     351.149  ops/s
[info] BigIntBenchmark.writeCirce    10000  thrpt    5      1418.141 ±      26.921  ops/s
[info] BigIntBenchmark.writeCirce   100000  thrpt    5        54.196 ±       1.848  ops/s
[info] BigIntBenchmark.writeCirce  1000000  thrpt    5         1.843 ±       0.018  ops/s
```

**Comment by plokhotnyuk** [comment]
BTW if you want a better option for representation of surrogate keys in JSON then consider `Long` for 64-bit keys, `UUID` for 128-bit keys, and Base64 encoding for longer keys instead of `BigInt` with decimal representation.

**Comment by travisbrown** [comment]
Fixed in #1217 and #1218.
